src/java/com/phenix/pct/PCT.java

```diff
@@ -627,82 +627,35 @@ protected static final int nextRandomInt() {
      * @param directory directory to delete
      * @throws IOException in case deletion is unsuccessful
      */
-    protected static void deleteDirectory(File directory) throws IOException {
-        if (!directory.exists()) {
+    protected void deleteDirectory(File directory) {
+        if (directory == null)
             return;
-        }
-
-        cleanDirectory(directory);
-
-        if (!directory.delete()) {
-            String message = "Unable to delete directory " + directory + ".";
-            throw new IOException(message);
-        }
-    }
-
-    /**
-     * Cleans a directory without deleting it.
-     * 
-     * @param directory directory to clean
-     * @throws IOException in case cleaning is unsuccessful
-     */
-    private static void cleanDirectory(File directory) throws IOException {
         if (!directory.exists()) {
-            String message = directory + " does not exist";
-            throw new IllegalArgumentException(message);
-        }
-
-        if (!directory.isDirectory()) {
-            String message = directory + " is not a directory";
-            throw new IllegalArgumentException(message);
+            return;
         }
 
-        File[] files = directory.listFiles();
-        if (files == null) { // null if security restricted
-            throw new IOException("Failed to list contents of " + directory);
-        }
-
-        IOException exception = null;
-        for (int zz = 0; zz < files.length; zz++) {
-            try {
-                forceDelete(files[zz]);
-            } catch (IOException ioe) {
-                exception = ioe;
-            }
+        for (File f : directory.listFiles()) {
+            deleteFile(f);
         }
+        try {
+            Files.deleteIfExists(directory.toPath());
+        } catch (IOException caught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)